backend/app/routes/admin_routes.py

```diff
@@ -55,12 +55,12 @@ def expire_credit(credit_id):
 
     user = User.query.filter_by(username=current_user.get('username')).first()
     credit = Credit.query.get(credit_id)
-    pc = PurchasedCredit.query.get(credit_id)
+    pc = PurchasedCredit.query.filter_by(credit_id=credit.id).first()
 
     if not credit:
         return jsonify({"message": "Credit not found"}), 404
     if not pc:
-        return jsonify({"message": "Credit can't be expired as it has not been sold yet"}), 400
+        return jsonify({"message": f"Credit can't be expired as it has not been sold yet, credit with B_ID {credit_id} is not found"}), 400
```

Comment:
what is B_ID?

Reply:
most probably means Backend_ID (as backend is 1 indexed and contract is 0)